### ISSUE TITLE
selection flicker

### ISSUE THREAD

**Issue by rjkroege** [rjkroege/edwood#56]
When text is selected, scrolling is slower. There is obviously too much work being done. I should make it better. In particular, there is no point in redrawing the selected text multiple times.

**Comment by rjkroege** [comment]
I spent a little bit of time trying to understand the scheduling behaviour between the draw library and `devdraw` program. It doesn't seem like `devdraw` has a particularly good way to express the two separate ideas of:

* permit subsequent pixel reads to depend on previous pixel writes
* show on screen at the next vblank all draw operations up to this point in the command stream.

The way that I simplified the selection painting code also overdraws.

**Comment by Superpat** [comment]
Not sure if it's the same thing, but under OpenBSD 6.4-current, I find Edwood to be noticeably slower than plan9port Acme when playing with selections.

**Comment by rjkroege** [comment]
Indeed. Fixing that is the point of the bug. I know why this is. I made a simplifying assumption in the implementation of `Frame` that was regrettable. There are a lot of opportunities to improve the `Frame` drawing code.

**Comment by rjkroege** [comment]
Also note several points in the code where I'm redrawing unnecessarily.
